### ISSUE TITLE
Allow launching the app with a custom wrapper script

### ISSUE THREAD

**Issue by PanAeon** [johnfactotum/foliate#701]
I run into a problem when I tried to package the app for the `NixOS`. The original packaging request is  [here](https://github.com/NixOS/nixpkgs/issues/108461).

In a nutshell the normal way of launching a gnome apps with `nix` is to provide small wrapper that will set all required paths for the system dependencies. `Foliate` atm requires the executable to be called exactly `com.github.johnfactotum.Foliate` in order to find its resources. As a consequence standard nix build fails as it renames the actual executable and calls it with the wrapper script instead.
I think this behaviour could be improved if only for the sake of flexibility.

**Proposed solution**
Change the `mason` build so the resources are fetched from `com.github.johnfactotum.Foliate` namespace independently from the executable name. 

**Additional context**
It looks like guys from `polari` mitigated the problem by providing a small C wrapper which includes the required resources.
[build: Use a small C wrapper ](https://gitlab.gnome.org/GNOME/polari/commit/f5dce05bb32b4409b32ff18ab5b7a71d27a02bfd) Not sure if this fix is applicable here, though.

**Comment by johnfactotum** [comment]
So, if I understand correctly, the problem is that GJS's package module gets the app's ID from the file name of the script?

If that's the case, maybe this should be fixed in GJS by allowing an additional parameter in `package.init()` and use that instead of the script's base name?

**Comment by PanAeon** [comment]
The thing is I don't know much about `GJS` at all))) From the [spec](https://gitlab.gnome.org/GNOME/gjs/-/blob/master/doc/Package/Specification.md):
```
Applications complying with this specification will have one application script, installed in ${prefix}/share/${package-name} (aka ${pkgdatadir}), and named as ${entry-point-name}, without any extension or mangling.
Optionally, one or more symlinks will be placed in ${bindir}, pointing to the appropriate script in ${pkgdatadir} and named in a fashion more suitable for command line usage (usually ${package-tarname}). Alternatively, a script that calls "gapplication launch ${package-name}" can be used.
```
So clearly the authors of the spec considered the possibility of an arbitrary name for the launch script, but right now it's hard for me to wrap my head around it.

To be honest this is more of a cosmetic issue atm, as the entry script can be easily patched to account for different app ID.

**Comment by johnfactotum** [comment]
> So clearly the authors of the spec considered the possibility of an arbitrary name for the launch script, but right now it's hard for me to wrap my head around it.

I think what it means here is that you can, for example, make a symlink called `foliate` in the `bin` direcotory, and then you don't have to type out the full name when using it from the command line. For example, this is done in the [Arch Linux package](https://github.com/archlinux/svntogit-community/blob/632cbc17382943efb6f7c2516e4028a4464a9f63/trunk/PKGBUILD#L33).

The [final destination of the symlink chain](https://gitlab.gnome.org/GNOME/gjs/-/blob/1.68.0/modules/script/package.js#L37-43), however, is [treated](https://gitlab.gnome.org/GNOME/gjs/-/blob/1.68.0/modules/script/package.js#L105) by GJS's Package module as the app's ID when [loading resource files](https://gitlab.gnome.org/GNOME/gjs/-/blob/1.68.0/modules/script/package.js#L159-163). So for the Package module to work, the real, non-symlink entry point script must be called `com.github.johnfactotum.Foliate`.

**Comment by PanAeon** [comment]
@johnfactotum I think the best decision at this point is to simply close this issue. I really hate when tons of issues are just clutter the bug tracker, although it might be just my adhd))
The patch in the NixOS codebase is just a small inconvienience, it shouldn't be a huge problem for other people with similar problems.

**Comment by johnfactotum** [comment]
This was fixed in 45e270b115f41573ac24e4ca027f3e8b7c7a7287. It doesn't use the `package` module anymore. It now loads the resource file directly from the app ID hardcoded in `main.js`.
